src/overlays/actors/ovl_Bg_Hidan_Firewall/z_bg_hidan_firewall.c

```diff
@@ -21,23 +22,239 @@ const ActorInit Bg_Hidan_Firewall_InitVars =
+    {
+        this->actor.draw = NULL;
+        this->actionFunc = BgHidanFirewall_Wait;
+    }
+    else
+    {
+        this->actor.params = 0;
+    }
+}
+
+void BgHidanFirewall_Collide(BgHidanFirewall* this, GlobalContext* globalCtx)
+{
+    s16 phi_a3;
+
+    if (func_8002E084(&this->actor, 0x4000) != 0)
+    {
+        phi_a3 = this->actor.shape.rot.y;
+    }
+    else
+    {
+        phi_a3 = this->actor.shape.rot.y + 0x8000;
+    }
+    func_8002F71C(globalCtx, this, 5.0f, phi_a3, 1.0f);
+}
+
+void BgHidanFirewall_ColliderFollowPlayer(BgHidanFirewall* this, GlobalContext* globalCtx)
+{
+    Player* player;
+    Vec3f sp30;
+    f32 temp_ret;
+    f32 sp28;
+    f32 phi_f0;
+
+    player = PLAYER;
+
+    func_8002DBD0(&this->actor, &sp30, &player->actor.posRot.pos); 
+    if (sp30.x < -70.0f)
+    {
+        sp30.x = -70.0f;
+    }
+    else
+    {
+        if (70.0f < sp30.x)
+        {
+            phi_f0 = 70.0f;
+        }
+        else
+        {
+            phi_f0 = sp30.x;
+        }
+        sp30.x = phi_f0;
+    }
+    if (this->actor.params == 0)
+    {
+        if (0.0f < sp30.z)
+        {
+            sp30.z = -25.0f;
+            this->actor.params = -1;
+        }
+        else
+        {
+            sp30.z = 25.0f;
+            this->actor.params = 1;
+        }
+    }
+    else
+    {
+        sp30.z = this->actor.params * 25.0f;
+    }
+    sp28 = Math_Sins(this->actor.shape.rot.y); 
+    temp_ret = Math_Coss(this->actor.shape.rot.y);
+    this->collider.dim.position.x = this->actor.posRot.pos.x + sp30.x * temp_ret + sp30.z * sp28;
+    this->collider.dim.position.z = this->actor.posRot.pos.z - sp30.x * sp28 + sp30.z * temp_ret;
+}
+
+void BgHidanFirewall_Update(BgHidanFirewall* this, GlobalContext* globalCtx)
+{
+    s32 pad1;
+    s32 pad2;
+
+    this->unk_150 = (this->unk_150 + 1) % 8;
+
+    if (this->collider.base.colliderFlags & 2)
+    {
+        this->collider.base.colliderFlags &= ~2;
+        BgHidanFirewall_Collide(this, globalCtx);
+    }
+
+    this->actionFunc(this, globalCtx);
+    if (this->actionFunc == BgHidanFirewall_Erupt)
+    {
+        BgHidanFirewall_ColliderFollowPlayer(this, globalCtx);
+        Actor_CollisionCheck_SetAT(globalCtx, &globalCtx->sub_11E60, &this->collider);
+        Actor_CollisionCheck_SetOT(globalCtx, &globalCtx->sub_11E60, &this->collider);
+        func_8002F974(&this->actor, 0x2034);
+    }
+}
+
+void BgHidanFirewall_Draw(BgHidanFirewall* this, GlobalContext* globalCtx)
+{
+    u32* temp;
+    u32 pad;
+    GraphicsContext* gfxCtx;
+    u32 pad1;
+    u32 pad2;
+    Gfx* gfx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Firewall/func_80886810.s")
+    gfxCtx = globalCtx->state.gfxCtx;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Hidan_Firewall/func_80886898.s")
+    func_800C6AC4(&gfx, globalCtx->state.gfxCtx, "../z_bg_hidan_firewall.c", 0x1C0);
+    gfxCtx->polyXlu.p = func_80093774(gfxCtx->polyXlu.p, 0x14);
```

Comment:
line numbers like this should be in decimal. you can recognize these debug functions that take line numbers at a glance by the filename string. Also this function name has changed, merge master to get the latest